<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>create training data</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">gene-level fine mapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">create training data</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">7/16/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-21
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>gene_level_fine_mapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200622code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200622)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200622code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200622)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/gene_level_fine_mapping/data/dt_annov_merged.csv
</td>
<td style="text-align:left;">
data/dt_annov_merged.csv
</td>
</tr>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/gene_level_fine_mapping/data/dt_neg_tss.RData
</td>
<td style="text-align:left;">
data/dt_neg_tss.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215genelevelfinemappingtree48b7920eb3d9e32b2746f9060037c3b2fad19cc1targetblank48b7920a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/tree/48b7920eb3d9e32b2746f9060037c3b2fad19cc1" target="_blank">48b7920</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215genelevelfinemappingtree48b7920eb3d9e32b2746f9060037c3b2fad19cc1targetblank48b7920a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/tree/48b7920eb3d9e32b2746f9060037c3b2fad19cc1" target="_blank">48b7920</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rproj.user/

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/create_training_set.Rmd</code>) and HTML (<code>docs/create_training_set.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/blob/48b7920eb3d9e32b2746f9060037c3b2fad19cc1/analysis/create_training_set.Rmd" target="_blank">48b7920</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2020-07-21
</td>
<td>
wflow_publish(“analysis/create_training_set.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<div id="annotation-result-summary" class="section level2">
<h2>annotation result summary</h2>
<pre class="r"><code>annov_merged = read.csv(&quot;/Users/nicholeyang/Desktop/gene_level_fine_mapping/data/dt_annov_merged.csv&quot;)</code></pre>
<pre class="r"><code>head(annov_merged)</code></pre>
<pre><code>  X         gene_id gene_name             variant_id      pip gene_chr
1 1 ENSG00000122477    LRRC39 chr1_100178174_A_G_b38 0.633113        1
2 2 ENSG00000162694     EXTL2 chr1_100895622_C_T_b38 0.573248        1
3 3 ENSG00000054523     KIF1B  chr1_10211630_C_G_b38 0.995686        1
4 4 ENSG00000240038     AMY2B chr1_103555063_C_T_b38 0.742527        1
5 5 ENSG00000226822 LINC02785 chr1_108045750_G_A_b38 0.999808        1
6 6 ENSG00000085491  SLC25A24 chr1_108199501_C_G_b38 0.809010        1
  gene_start  gene_end gene_width tss_chr tss_position tss_dist_to_snp
1  100148449 100178273      29825       1    100178215              41
2  100872372 100895179      22808       1    100895998             376
3   10210805  10381603     170799       1     10211616              14
4  103553815 103579534      25720       1    103554700             363
5  108040263 108076020      35758      NA           NA              NA
6  108134043 108200849      66807       1    108200358             857
    snp_func       snp_refgene
1       UTR5            LRRC39
2   upstream     EXTL2;SLC30A7
3       UTR5             KIF1B
4  UTR5;UTR3       AMY2B;RNPC3
5 intergenic VAV3-AS1;SLC25A24
6   intronic          SLC25A24
                                                                                refgene_detail
1 NM_001256385:c.-9658C&gt;T;NM_001256386:c.-9658C&gt;T;NM_001256387:c.-9658C&gt;T;NM_144620:c.-9658C&gt;T
2                                                                                     dist=468
3                                                                     NM_001365952:c.-20699G&gt;C
4                                                     NM_020978:c.-16540T&gt;C;NM_017619:c.*42T&gt;C
5                                                                        dist=51143;dist=88293
6                                                                                            .</code></pre>
<div id="summarizing-by-categories" class="section level3">
<h3>summarizing by categories</h3>
<pre class="r"><code>categ = unique(annov_merged$snp_func)
categ</code></pre>
<pre><code> [1] UTR5                upstream            UTR5;UTR3          
 [4] intergenic          intronic            .                  
 [7] ncRNA_intronic      exonic              ncRNA_exonic       
[10] downstream          UTR3                upstream;downstream
[13] ncRNA_splicing      splicing           
14 Levels: . downstream exonic intergenic intronic ... UTR5;UTR3</code></pre>
<pre class="r"><code>obs = rep(NA, length(categ))

for (i in 1:length(obs)){
  obs[i] = sum(annov_merged$snp_func == categ[i])
}

barplot(obs, names.arg = c(as.character(categ)), cex.names=0.7, las = 2, ylab = &#39;counts&#39;)</code></pre>
<p><img src="figure/create_training_set.Rmd/unnamed-chunk-4-1.png" width="672" style="display: block; margin: auto;" /></p>
</div>
</div>
<div id="create-features-utr5utr3exonintronupstream" class="section level2">
<h2>create features: UTR5/UTR3/exon/intron/upstream</h2>
<ol style="list-style-type: decimal">
<li>Remove feature categories: ‘UTR5;UTR3’, ‘ncRNA_intronic’, ‘ncRNA_exonic’, ‘upstream;downstream’ for simplification</li>
<li>Remove ‘.’, which means no annotation found</li>
</ol>
<pre class="r"><code>train = annov_merged[annov_merged$snp_func != &#39;UTR5;UTR3&#39; &amp; annov_merged$snp_func != &#39;ncRNA_intronic&#39; &amp;
                        annov_merged$snp_func != &#39;ncRNA_exonic&#39; &amp; annov_merged$snp_func != &#39;upstream;downstream&#39; &amp;
                     annov_merged$snp_func != &#39;.&#39;, ]


## some snps have 2 reference genes 
refgenes = strsplit(as.character(train$snp_refgene), &#39;;&#39;)   
train$refgene1 = unlist(lapply(refgenes, function(x) x[1]))
train$refgene2 = unlist(lapply(refgenes, function(x) x[2]))
train$refgene2[is.na(train$refgene2)] = &#39;999&#39;


train2 = train[, c(&#39;gene_id&#39;,&#39;gene_name&#39;, &#39;variant_id&#39;, &#39;snp_func&#39;, 
                   &#39;snp_refgene&#39;, &#39;refgene_detail&#39;, &#39;refgene1&#39;, &#39;refgene2&#39;, &#39;tss_dist_to_snp&#39;)]</code></pre>
</div>
<div id="two-situations-whether-associated-genes-reference-genes" class="section level2">
<h2>two situations: whether associated genes = reference genes</h2>
<pre class="r"><code>## 1. gene-snp pairs where associated gene = reference gene
sub_train1 = train2[as.character(train2$gene_name) == train2$refgene1 | as.character(train2$gene_name) == train2$refgene2, ]


## 2. gene-snp pairs where associated gene != reference gene
sub_train2 = train2[as.character(train2$gene_name) != train2$refgene1 &amp; 
                      as.character(train2$gene_name) != train2$refgene2, ]</code></pre>
</div>
<div id="create-positive-training-set" class="section level2">
<h2>create positive training set</h2>
<pre class="r"><code>pos1 = sub_train1
pos1$UTR5 = ifelse(pos1$snp_func == &#39;UTR5&#39;, 1, 0)
pos1$UTR3 = ifelse(pos1$snp_func == &#39;UTR3&#39;, 1, 0)
pos1$exon = ifelse(pos1$snp_func == &#39;exonic&#39;, 1, 0)
pos1$intron = ifelse(pos1$snp_func == &#39;intronic&#39;, 1, 0)
pos1$upstream = ifelse(pos1$snp_func == &#39;upstream&#39;, 1, 0)

pos2 = sub_train2
pos2$UTR5 = rep(0, dim(pos2)[1])
pos2$UTR3 = rep(0, dim(pos2)[1])
pos2$exon = rep(0, dim(pos2)[1])
pos2$intron = rep(0, dim(pos2)[1])
pos2$upstream = rep(0, dim(pos2)[1])

train_pos = rbind(pos1, pos2)
train_pos$y = rep(1, dim(train_pos)[1])</code></pre>
</div>
<div id="create-negative-training-set-two-sources" class="section level2">
<h2>create negative training set: two sources</h2>
<ol style="list-style-type: decimal">
<li><p>In positive set, for a snp where associated gene is not the reference gene. Then reference gene-snp will be the gene-snp pair.</p></li>
<li><p>the genes within 1mb of all the unique snps in the positive set.</p></li>
</ol>
<pre class="r"><code>## source1: e.g. geneA associated with snpA, but snpA reference to geneB. geneB-snpA is a negative gene-snp pair here

neg1 = sub_train2
neg1$UTR5 = ifelse(neg1$snp_func == &#39;UTR5&#39;, 1, 0)
neg1$UTR3 = ifelse(neg1$snp_func == &#39;UTR3&#39;, 1, 0)
neg1$exon = ifelse(neg1$snp_func == &#39;exonic&#39;, 1, 0)
neg1$intron = ifelse(neg1$snp_func == &#39;intronic&#39;, 1, 0)
neg1$upstream = ifelse(neg1$snp_func == &#39;upstream&#39;, 1, 0)

neg1$y = rep(0, dim(neg1)[1])</code></pre>
<pre class="r"><code>load(&quot;/Users/nicholeyang/Desktop/gene_level_fine_mapping/data/dt_neg_tss.RData&quot;)
head(dt_neg_tss)</code></pre>
<pre><code>       gene_name seqnames     start       end  width strand.x type
37846       DPM1       20  50934867  50958555  23689        - gene
120661     SCYL3        1 169849631 169894267  44637        - gene
120665     SCYL3        1 169849631 169894267  44637        - gene
120663     SCYL3        1 169849631 169894267  44637        - gene
15473   C1orf112        1 169662007 169854080 192074        + gene
15472   C1orf112        1 169662007 169854080 192074        + gene
             gene_id.x   gene_biotype                 var_id   SNP_loc
37846  ENSG00000000419 protein_coding chr20_50794732_A_G_b38  50794732
120661 ENSG00000000457 protein_coding chr1_169661963_G_A_b38 169661963
120665 ENSG00000000457 protein_coding chr1_169605649_C_T_b38 169605649
120663 ENSG00000000457 protein_coding chr1_169443797_C_T_b38 169443797
15473  ENSG00000000460 protein_coding chr1_169891332_G_A_b38 169891332
15472  ENSG00000000460 protein_coding chr1_169605649_C_T_b38 169605649
             gene_id.y chr tss_position strand.y dist_to_snp
37846  ENST00000371588   2     50958550        -      163818
120661 ENST00000367770   1    169888888        -      226925
120665 ENST00000367770   1    169888888        -      283239
120663 ENST00000367770   1    169888888        -      445091
15473  ENST00000413811   1    169795921        +       95411
15472  ENST00000496973   1    169795043        +      189394</code></pre>
<pre class="r"><code>## find the genes within 1mb for the unique snp set. 
unique_snp = unique(as.character(train2$variant_id))
indx = unlist(lapply(unique_snp, function(x) which(dt_neg_tss$var_id == x)))
neg2 = dt_neg_tss[indx, ]</code></pre>
<pre class="r"><code>## remove the gene-snp pairs already in the negative set1. 
neg2_remove_indx = c()

for (i in 1:dim(sub_train2)[1]){
  
  gene_name = sub_train2[i, ]$refgene1
  snp_id = sub_train2[i, ]$variant_id
  
  indx = which(as.character(neg2$gene_name) == as.character(gene_name) &amp; as.character(neg2$var_id) == as.character(snp_id))
  neg2_remove_indx = c(neg2_remove_indx, indx)
}</code></pre>
<pre class="r"><code>neg2 = neg2[-neg2_remove_indx, ]
neg2$UTR5 = rep(0, dim(neg2)[1])
neg2$UTR3 = rep(0, dim(neg2)[1])
neg2$exon = rep(0, dim(neg2)[1])
neg2$intron = rep(0, dim(neg2)[1])
neg2$upstream = rep(0, dim(neg2)[1])
neg2$y = rep(0, dim(neg2)[1])</code></pre>
<pre class="r"><code>train_pos_all = train_pos[, c(&#39;gene_name&#39;, &#39;variant_id&#39;, &quot;UTR5&quot;, &quot;UTR3&quot;, &quot;exon&quot;, 
                              &quot;intron&quot;, &quot;upstream&quot;, &#39;tss_dist_to_snp&#39;, &quot;y&quot;)]


sub_train_neg1 = neg1[, c(&#39;snp_refgene&#39;, &#39;variant_id&#39;, &quot;UTR5&quot;, &quot;UTR3&quot;, &quot;exon&quot;, 
                              &quot;intron&quot;, &quot;upstream&quot;, &#39;tss_dist_to_snp&#39;, &quot;y&quot;)]

colnames(sub_train_neg1) = c(&#39;gene_name&#39;, &#39;variant_id&#39;, &quot;UTR5&quot;, &quot;UTR3&quot;, &quot;exon&quot;, 
                              &quot;intron&quot;, &quot;upstream&quot;, &#39;tss_dist_to_snp&#39;, &quot;y&quot;)

sub_train_neg2 = neg2[, c(&#39;gene_name&#39;, &#39;var_id&#39;, &quot;UTR5&quot;, &quot;UTR3&quot;, &quot;exon&quot;, 
                              &quot;intron&quot;, &quot;upstream&quot;, &#39;dist_to_snp&#39;, &quot;y&quot;)]

colnames(sub_train_neg2) = c(&#39;gene_name&#39;, &#39;variant_id&#39;, &quot;UTR5&quot;, &quot;UTR3&quot;, &quot;exon&quot;, 
                              &quot;intron&quot;, &quot;upstream&quot;, &#39;tss_dist_to_snp&#39;, &quot;y&quot;)


train_neg_all = rbind(sub_train_neg1, sub_train_neg2)

# save(train_pos_all, train_neg_all, file = &#39;training.RData&#39;)</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()</code></pre>
<pre><code>R version 3.6.3 (2020-02-29)
Platform: x86_64-apple-darwin15.6.0 (64-bit)
Running under: macOS Catalina 10.15.5

Matrix products: default
BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib

locale:
[1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8

attached base packages:
[1] stats     graphics  grDevices utils     datasets  methods   base     

other attached packages:
[1] workflowr_1.6.1

loaded via a namespace (and not attached):
 [1] Rcpp_1.0.4      rprojroot_1.3-2 digest_0.6.25   later_1.0.0    
 [5] R6_2.4.1        backports_1.1.5 git2r_0.26.1    magrittr_1.5   
 [9] evaluate_0.14   highr_0.8       stringi_1.4.6   rlang_0.4.5    
[13] fs_1.3.2        promises_1.1.0  whisker_0.4     rmarkdown_2.1  
[17] tools_3.6.3     stringr_1.4.0   glue_1.3.2      httpuv_1.5.2   
[21] xfun_0.12       yaml_2.2.1      compiler_3.6.3  htmltools_0.4.0
[25] knitr_1.28     </code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
