<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="Yunqi Yang" />


<title>loci_annotation</title>

<script src="site_libs/jquery-1.11.3/jquery.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/cosmo.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<script src="site_libs/jqueryui-1.11.4/jquery-ui.min.js"></script>
<link href="site_libs/tocify-1.9.1/jquery.tocify.css" rel="stylesheet" />
<script src="site_libs/tocify-1.9.1/jquery.tocify.js"></script>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<link rel="icon" href="https://github.com/workflowr/workflowr-assets/raw/master/img/reproducible.png">
<!-- Add a small amount of space between sections. -->
<style type="text/css">
div.section {
  padding-top: 12px;
}
</style>

<style type="text/css">code{white-space: pre;}</style>
<style type="text/css">
  pre:not([class]) {
    background-color: white;
  }
</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>



<style type="text/css">
h1 {
  font-size: 34px;
}
h1.title {
  font-size: 38px;
}
h2 {
  font-size: 30px;
}
h3 {
  font-size: 24px;
}
h4 {
  font-size: 18px;
}
h5 {
  font-size: 16px;
}
h6 {
  font-size: 12px;
}
.table th:not([align]) {
  text-align: left;
}
</style>




<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
code {
  color: inherit;
  background-color: rgba(0, 0, 0, 0.04);
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
</style>


<style type="text/css">
/* padding for bootstrap navbar */
body {
  padding-top: 51px;
  padding-bottom: 40px;
}
/* offset scroll position for anchor links (for fixed navbar)  */
.section h1 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h2 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h3 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h4 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h5 {
  padding-top: 56px;
  margin-top: -56px;
}
.section h6 {
  padding-top: 56px;
  margin-top: -56px;
}
.dropdown-submenu {
  position: relative;
}
.dropdown-submenu>.dropdown-menu {
  top: 0;
  left: 100%;
  margin-top: -6px;
  margin-left: -1px;
  border-radius: 0 6px 6px 6px;
}
.dropdown-submenu:hover>.dropdown-menu {
  display: block;
}
.dropdown-submenu>a:after {
  display: block;
  content: " ";
  float: right;
  width: 0;
  height: 0;
  border-color: transparent;
  border-style: solid;
  border-width: 5px 0 5px 5px;
  border-left-color: #cccccc;
  margin-top: 5px;
  margin-right: -10px;
}
.dropdown-submenu:hover>a:after {
  border-left-color: #ffffff;
}
.dropdown-submenu.pull-left {
  float: none;
}
.dropdown-submenu.pull-left>.dropdown-menu {
  left: -100%;
  margin-left: 10px;
  border-radius: 6px 0 6px 6px;
}
</style>

<script>
// manage active state of menu based on current page
$(document).ready(function () {
  // active menu anchor
  href = window.location.pathname
  href = href.substr(href.lastIndexOf('/') + 1)
  if (href === "")
    href = "index.html";
  var menuAnchor = $('a[href="' + href + '"]');

  // mark it active
  menuAnchor.parent().addClass('active');

  // if it's got a parent navbar menu mark it active as well
  menuAnchor.closest('li.dropdown').addClass('active');
});
</script>

<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  background: white;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->



<style type="text/css">

#TOC {
  margin: 25px 0px 20px 0px;
}
@media (max-width: 768px) {
#TOC {
  position: relative;
  width: 100%;
}
}

@media print {
.toc-content {
  /* see https://github.com/w3c/csswg-drafts/issues/4434 */
  float: right;
}
}

.toc-content {
  padding-left: 30px;
  padding-right: 40px;
}

div.main-container {
  max-width: 1200px;
}

div.tocify {
  width: 20%;
  max-width: 260px;
  max-height: 85%;
}

@media (min-width: 768px) and (max-width: 991px) {
  div.tocify {
    width: 25%;
  }
}

@media (max-width: 767px) {
  div.tocify {
    width: 100%;
    max-width: none;
  }
}

.tocify ul, .tocify li {
  line-height: 20px;
}

.tocify-subheader .tocify-item {
  font-size: 0.90em;
}

.tocify .list-group-item {
  border-radius: 0px;
}


</style>



</head>

<body>


<div class="container-fluid main-container">


<!-- setup 3col/9col grid for toc_float and main content  -->
<div class="row-fluid">
<div class="col-xs-12 col-sm-4 col-md-3">
<div id="TOC" class="tocify">
</div>
</div>

<div class="toc-content col-xs-12 col-sm-8 col-md-9">




<div class="navbar navbar-default  navbar-fixed-top" role="navigation">
  <div class="container">
    <div class="navbar-header">
      <button type="button" class="navbar-toggle collapsed" data-toggle="collapse" data-target="#navbar">
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
        <span class="icon-bar"></span>
      </button>
      <a class="navbar-brand" href="index.html">gene-level fine mapping</a>
    </div>
    <div id="navbar" class="navbar-collapse collapse">
      <ul class="nav navbar-nav">
        <li>
  <a href="index.html">Home</a>
</li>
<li>
  <a href="about.html">About</a>
</li>
<li>
  <a href="license.html">License</a>
</li>
      </ul>
      <ul class="nav navbar-nav navbar-right">
        
      </ul>
    </div><!--/.nav-collapse -->
  </div><!--/.container -->
</div><!--/.navbar -->

<div class="fluid-row" id="header">



<h1 class="title toc-ignore">loci_annotation</h1>
<h4 class="author">Yunqi Yang</h4>
<h4 class="date">7/6/2020</h4>

</div>


<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-report" data-toggle="collapse" data-target="#workflowr-report">
<span class="glyphicon glyphicon-list" aria-hidden="true"></span> workflowr <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span>
</button>
</p>
<div id="workflowr-report" class="collapse">
<ul class="nav nav-tabs">
<li class="active">
<a data-toggle="tab" href="#summary">Summary</a>
</li>
<li>
<a data-toggle="tab" href="#checks"> Checks <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> </a>
</li>
<li>
<a data-toggle="tab" href="#versions">Past versions</a>
</li>
</ul>
<div class="tab-content">
<div id="summary" class="tab-pane fade in active">
<p>
<strong>Last updated:</strong> 2020-07-13
</p>
<p>
<strong>Checks:</strong> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> 6 <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> 1
</p>
<p>
<strong>Knit directory:</strong> <code>gene-level fine mapping/</code> <span class="glyphicon glyphicon-question-sign" aria-hidden="true" title="This is the local directory in which the code in this file was executed."> </span>
</p>
<p>
This reproducible <a href="http://rmarkdown.rstudio.com">R Markdown</a> analysis was created with <a
  href="https://github.com/jdblischak/workflowr">workflowr</a> (version 1.6.1). The <em>Checks</em> tab describes the reproducibility checks that were applied when the results were created. The <em>Past versions</em> tab lists the development history.
</p>
<hr>
</div>
<div id="checks" class="tab-pane fade">
<div id="workflowr-checks" class="panel-group">
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRMarkdownfilestronguptodate"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>R Markdown file:</strong> up-to-date </a>
</p>
</div>
<div id="strongRMarkdownfilestronguptodate" class="panel-collapse collapse">
<div class="panel-body">
<p>Great! Since the R Markdown file has been committed to the Git repository, you know the exact version of the code that produced these results.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongEnvironmentstrongempty"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Environment:</strong> empty </a>
</p>
</div>
<div id="strongEnvironmentstrongempty" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! The global environment was empty. Objects defined in the global environment can affect the analysis in your R Markdown file in unknown ways. For reproduciblity it’s best to always run the code in an empty environment.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSeedstrongcodesetseed20200622code"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Seed:</strong> <code>set.seed(20200622)</code> </a>
</p>
</div>
<div id="strongSeedstrongcodesetseed20200622code" class="panel-collapse collapse">
<div class="panel-body">
<p>The command <code>set.seed(20200622)</code> was run prior to running the code in the R Markdown file. Setting a seed ensures that any results that rely on randomness, e.g. subsampling or permutations, are reproducible.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongSessioninformationstrongrecorded"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Session information:</strong> recorded </a>
</p>
</div>
<div id="strongSessioninformationstrongrecorded" class="panel-collapse collapse">
<div class="panel-body">
<p>Great job! Recording the operating system, R version, and package versions is critical for reproducibility.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongCachestrongnone"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Cache:</strong> none </a>
</p>
</div>
<div id="strongCachestrongnone" class="panel-collapse collapse">
<div class="panel-body">
<p>Nice! There were no cached chunks for this analysis, so you can be confident that you successfully produced the results during this run.</p>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongFilepathsstrongabsolute"> <span class="glyphicon glyphicon-exclamation-sign text-danger" aria-hidden="true"></span> <strong>File paths:</strong> absolute </a>
</p>
</div>
<div id="strongFilepathsstrongabsolute" class="panel-collapse collapse">
<div class="panel-body">
<p>
Using absolute paths to the files within your workflowr project makes it difficult for you and others to run your code on a different machine. Change the absolute path(s) below to the suggested relative path(s) to make your code more reproducible.
</p>
<table class="table table-condensed table-hover">
<thead>
<tr>
<th style="text-align:left;">
absolute
</th>
<th style="text-align:left;">
relative
</th>
</tr>
</thead>
<tbody>
<tr>
<td style="text-align:left;">
/Users/nicholeyang/Desktop/Rotation2/gene-level fine mapping/data/dt_pos_processed.RData
</td>
<td style="text-align:left;">
data/dt_pos_processed.RData
</td>
</tr>
</tbody>
</table>
</div>
</div>
</div>
<div class="panel panel-default">
<div class="panel-heading">
<p class="panel-title">
<a data-toggle="collapse" data-parent="#workflowr-checks" href="#strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215genelevelfinemappingtree1829293d2cd1ff1b31fff3f3e69320c31b6e5a94targetblank1829293a"> <span class="glyphicon glyphicon-ok text-success" aria-hidden="true"></span> <strong>Repository version:</strong> <a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/tree/1829293d2cd1ff1b31fff3f3e69320c31b6e5a94" target="_blank">1829293</a> </a>
</p>
</div>
<div id="strongRepositoryversionstrongahrefhttpsgithubcomyunqiyang0215genelevelfinemappingtree1829293d2cd1ff1b31fff3f3e69320c31b6e5a94targetblank1829293a" class="panel-collapse collapse">
<div class="panel-body">
<p>
Great! You are using Git for version control. Tracking code development and connecting the code version to the results is critical for reproducibility.
</p>
<p>
The results in this page were generated with repository version <a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/tree/1829293d2cd1ff1b31fff3f3e69320c31b6e5a94" target="_blank">1829293</a>. See the <em>Past versions</em> tab to see a history of the changes made to the R Markdown and HTML files.
</p>
<p>
Note that you need to be careful to ensure that all relevant files for the analysis have been committed to Git prior to generating the results (you can use <code>wflow_publish</code> or <code>wflow_git_commit</code>). workflowr only checks the R Markdown file, but you know if there are other scripts or data files that it depends on. Below is the status of the Git repository when the results were generated:
</p>
<pre><code>
Ignored files:
    Ignored:    .DS_Store
    Ignored:    .Rhistory
    Ignored:    .Rproj.user/
    Ignored:    analysis/.DS_Store
    Ignored:    analysis/.RData
    Ignored:    analysis/.Rhistory
    Ignored:    data/.DS_Store

Untracked files:
    Untracked:  analysis/annovar.Rmd
    Untracked:  analysis/dt_pos3.RData
    Untracked:  data/Homo_sapiens.GRCh37.87.gtf
    Untracked:  data/Homo_sapiens.GRCh37.87.gtf.gz
    Untracked:  data/README (1)
    Untracked:  data/SNPs_causal.RData
    Untracked:  data/dt_pos_processed.RData
    Untracked:  dt_pos3.RData
    Untracked:  dt_pos_processed.RData

Unstaged changes:
    Modified:   analysis/eqtl_process.Rmd

</code></pre>
<p>
Note that any generated files, e.g. HTML, png, CSS, etc., are not included in this status report because it is ok for generated content to have uncommitted changes.
</p>
</div>
</div>
</div>
</div>
<hr>
</div>
<div id="versions" class="tab-pane fade">

<p>
These are the previous versions of the repository in which changes were made to the R Markdown (<code>analysis/gene_annotation.Rmd</code>) and HTML (<code>docs/gene_annotation.html</code>) files. If you’ve configured a remote Git repository (see <code>?wflow_git_remote</code>), click on the hyperlinks in the table below to view the files as they were in that past version.
</p>
<div class="table-responsive">
<table class="table table-condensed table-hover">
<thead>
<tr>
<th>
File
</th>
<th>
Version
</th>
<th>
Author
</th>
<th>
Date
</th>
<th>
Message
</th>
</tr>
</thead>
<tbody>
<tr>
<td>
Rmd
</td>
<td>
<a href="https://github.com/yunqiyang0215/gene_level_fine_mapping/blob/1829293d2cd1ff1b31fff3f3e69320c31b6e5a94/analysis/gene_annotation.Rmd" target="_blank">1829293</a>
</td>
<td>
yunqiyang0215
</td>
<td>
2020-07-13
</td>
<td>
wflow_publish(“analysis/gene_annotation.Rmd”)
</td>
</tr>
</tbody>
</table>
</div>
<hr>
</div>
</div>
</div>
<pre class="r"><code>
setwd(&#39;/Library/Frameworks/R.framework/Versions/3.6/Resources/library/&#39;)
devtools::install(pkg = &quot;XML&quot;)

if (!requireNamespace(&quot;BiocManager&quot;, quietly = TRUE))
    install.packages(&quot;BiocManager&quot;)
BiocManager::install()

BiocManager::install(c(&quot;GenomicFeatures&quot;, &quot;AnnotationDbi&quot;, &#39;rtracklayer&#39;))
</code></pre>
<pre class="r"><code>
library(rtracklayer)
# Loading required package: GenomicRanges
# Loading required package: stats4
# Loading required package: BiocGenerics
# Loading required package: parallel
# 
# Attaching package: &#39;BiocGenerics&#39;
# The following objects are masked from &#39;package:parallel&#39;:
# 
#     clusterApply, clusterApplyLB, clusterCall, clusterEvalQ,
#     clusterExport, clusterMap, parApply, parCapply, parLapply,
#     parLapplyLB, parRapply, parSapply, parSapplyLB
# The following objects are masked from &#39;package:stats&#39;:
# 
#     IQR, mad, sd, var, xtabs
# The following objects are masked from &#39;package:base&#39;:
# 
#     anyDuplicated, append, as.data.frame, basename, cbind, colnames,
#     dirname, do.call, duplicated, eval, evalq, Filter, Find, get, grep,
#     grepl, intersect, is.unsorted, lapply, Map, mapply, match, mget,
#     order, paste, pmax, pmax.int, pmin, pmin.int, Position, rank,
#     rbind, Reduce, rownames, sapply, setdiff, sort, table, tapply,
#     union, unique, unsplit, which, which.max, which.min
# Loading required package: S4Vectors
# 
# Attaching package: &#39;S4Vectors&#39;
# The following object is masked from &#39;package:base&#39;:
# 
#     expand.grid
# Loading required package: IRanges
# Loading required package: GenomeInfoDb</code></pre>
<pre class="r"><code>
# load fine-mapping positive set
load(&quot;/Users/nicholeyang/Desktop/Rotation2/gene-level fine mapping/data/dt_pos_processed.RData&quot;)

dt = import(&quot;/Users/nicholeyang/Desktop/Rotation2/data/Homo_sapiens.GRCh38.100.gtf&quot;)
dt = as.data.frame(dt)</code></pre>
<pre class="r"><code>
head(dt_pos)
#                   gene rank             variant_id      pip log10_abf
# 5   ENSG00000000457.13    1 chr1_169891332_G_A_b38 0.939551    20.142
# 43  ENSG00000000460.16    1 chr1_169661963_G_A_b38 0.580082     5.656
# 307  ENSG00000001561.6    1  chr6_46130021_C_G_b38 0.657995     5.590
# 308  ENSG00000001561.6    2 chr6_46129743_G_GT_b38 0.513630    22.020
# 418  ENSG00000001629.9    1  chr7_92245996_C_T_b38 0.520558     7.253
# 594 ENSG00000002016.17    1   chr12_949572_C_G_b38 0.522047    23.668
#     cluster_id SNP_chr   SNP_loc         gene_id
# 5            1       1 169891332 ENSG00000000457
# 43           1       1 169661963 ENSG00000000460
# 307          2       6  46130021 ENSG00000001561
# 308          3       6  46129743 ENSG00000001561
# 418          1       7  92245996 ENSG00000001629
# 594          2      12    949572 ENSG00000002016</code></pre>
<div id="find-the-gene-coordinate-in-the-positive-set" class="section level2">
<h2>find the gene coordinate in the positive set</h2>
<pre class="r"><code>
## subset to type = gene
hsgene = dt[dt$type == &#39;gene&#39;, ]
head(hsgene)
#    seqnames start   end width strand  source type score phase         gene_id
# 1         1 11869 14409  2541      +  havana gene    NA    NA ENSG00000223972
# 13        1 14404 29570 15167      -  havana gene    NA    NA ENSG00000227232
# 26        1 17369 17436    68      - mirbase gene    NA    NA ENSG00000278267
# 29        1 29554 31109  1556      +  havana gene    NA    NA ENSG00000243485
# 37        1 30366 30503   138      + mirbase gene    NA    NA ENSG00000284332
# 40        1 34554 36081  1528      -  havana gene    NA    NA ENSG00000237613
#    gene_version   gene_name gene_source                       gene_biotype
# 1             5     DDX11L1      havana transcribed_unprocessed_pseudogene
# 13            5      WASH7P      havana             unprocessed_pseudogene
# 26            1   MIR6859-1     mirbase                              miRNA
# 29            5 MIR1302-2HG      havana                             lncRNA
# 37            1   MIR1302-2     mirbase                              miRNA
# 40            2     FAM138A      havana                             lncRNA
#    transcript_id transcript_version transcript_name transcript_source
# 1           &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
# 13          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
# 26          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
# 29          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
# 37          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
# 40          &lt;NA&gt;               &lt;NA&gt;            &lt;NA&gt;              &lt;NA&gt;
#    transcript_biotype  tag transcript_support_level exon_number exon_id
# 1                &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
# 13               &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
# 26               &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
# 29               &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
# 37               &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
# 40               &lt;NA&gt; &lt;NA&gt;                     &lt;NA&gt;        &lt;NA&gt;    &lt;NA&gt;
#    exon_version protein_id protein_version ccds_id
# 1          &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;
# 13         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;
# 26         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;
# 29         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;
# 37         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;
# 40         &lt;NA&gt;       &lt;NA&gt;            &lt;NA&gt;    &lt;NA&gt;</code></pre>
<pre class="r"><code>
## obtain the gene coordinates
pos_gene_coor = hsgene[which(hsgene$gene_id == dt_pos$gene_id[1]), c(&#39;gene_id&#39;,&#39;gene_name&#39;,&#39;seqnames&#39;, &#39;start&#39;, &#39;end&#39;, &#39;width&#39;)]

for (i in 2:length(dt_pos$gene_id)){
  pos_gene_coor = rbind(pos_gene_coor, hsgene[which(hsgene$gene_id == dt_pos$gene_id[i]), c(&#39;gene_id&#39;,&#39;gene_name&#39;,&#39;seqnames&#39;, &#39;start&#39;, &#39;end&#39;, &#39;width&#39;)])
}</code></pre>
<pre class="r"><code>

dt_pos2 = merge(dt_pos, pos_gene_coor, by = &#39;gene_id&#39;)
dt_pos3 = dt_pos2[!duplicated(dt_pos2), ]

head(dt_pos3)
#           gene_id               gene rank             variant_id      pip
# 1 ENSG00000000457 ENSG00000000457.13    1 chr1_169891332_G_A_b38 0.939551
# 2 ENSG00000000460 ENSG00000000460.16    1 chr1_169661963_G_A_b38 0.580082
# 3 ENSG00000001561  ENSG00000001561.6    2 chr6_46129743_G_GT_b38 0.513630
# 5 ENSG00000001561  ENSG00000001561.6    1  chr6_46130021_C_G_b38 0.657995
# 7 ENSG00000001629  ENSG00000001629.9    1  chr7_92245996_C_T_b38 0.520558
# 8 ENSG00000002016 ENSG00000002016.17    1   chr12_949572_C_G_b38 0.522047
#   log10_abf cluster_id SNP_chr   SNP_loc gene_name seqnames     start       end
# 1    20.142          1       1 169891332     SCYL3        1 169849631 169894267
# 2     5.656          1       1 169661963  C1orf112        1 169662007 169854080
# 3    22.020          3       6  46129743     ENPP4        6  46129989  46146688
# 5     5.590          2       6  46130021     ENPP4        6  46129989  46146688
# 7     7.253          1       7  92245996    ANKIB1        7  92245974  92401383
# 8    23.668          2      12    949572     RAD52       12    911736    990053
#    width
# 1  44637
# 2 192074
# 3  16700
# 5  16700
# 7 155410
# 8  78318</code></pre>
<pre class="r"><code>
save(dt_pos3, file = &#39;dt_pos3.RData&#39;)
</code></pre>
<br>
<p>
<button type="button" class="btn btn-default btn-workflowr btn-workflowr-sessioninfo" data-toggle="collapse" data-target="#workflowr-sessioninfo" style="display: block;">
<span class="glyphicon glyphicon-wrench" aria-hidden="true"></span> Session information
</button>
</p>
<div id="workflowr-sessioninfo" class="collapse">
<pre class="r"><code>sessionInfo()
# R version 3.6.3 (2020-02-29)
# Platform: x86_64-apple-darwin15.6.0 (64-bit)
# Running under: macOS Catalina 10.15.5
# 
# Matrix products: default
# BLAS:   /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRblas.0.dylib
# LAPACK: /Library/Frameworks/R.framework/Versions/3.6/Resources/lib/libRlapack.dylib
# 
# locale:
# [1] en_US.UTF-8/en_US.UTF-8/en_US.UTF-8/C/en_US.UTF-8/en_US.UTF-8
# 
# attached base packages:
# [1] parallel  stats4    stats     graphics  grDevices utils     datasets 
# [8] methods   base     
# 
# other attached packages:
# [1] rtracklayer_1.46.0   GenomicRanges_1.38.0 GenomeInfoDb_1.22.1 
# [4] IRanges_2.20.2       S4Vectors_0.24.4     BiocGenerics_0.32.0 
# [7] workflowr_1.6.1     
# 
# loaded via a namespace (and not attached):
#  [1] Rcpp_1.0.4                  compiler_3.6.3             
#  [3] later_1.0.0                 git2r_0.26.1               
#  [5] highr_0.8                   XVector_0.26.0             
#  [7] bitops_1.0-6                tools_3.6.3                
#  [9] zlibbioc_1.32.0             digest_0.6.25              
# [11] lattice_0.20-40             evaluate_0.14              
# [13] rlang_0.4.5                 Matrix_1.2-18              
# [15] DelayedArray_0.12.3         yaml_2.2.1                 
# [17] xfun_0.12                   GenomeInfoDbData_1.2.2     
# [19] stringr_1.4.0               knitr_1.28                 
# [21] Biostrings_2.54.0           fs_1.3.2                   
# [23] grid_3.6.3                  rprojroot_1.3-2            
# [25] Biobase_2.46.0              glue_1.3.2                 
# [27] R6_2.4.1                    XML_3.99-0.3               
# [29] BiocParallel_1.20.1         rmarkdown_2.1              
# [31] magrittr_1.5                whisker_0.4                
# [33] matrixStats_0.56.0          GenomicAlignments_1.22.1   
# [35] backports_1.1.5             Rsamtools_2.2.3            
# [37] promises_1.1.0              htmltools_0.4.0            
# [39] SummarizedExperiment_1.16.1 httpuv_1.5.2               
# [41] stringi_1.4.6               RCurl_1.98-1.2</code></pre>
</div>
</div>


<!-- Adjust MathJax settings so that all math formulae are shown using
TeX fonts only; see
http://docs.mathjax.org/en/latest/configuration.html.  This will make
the presentation more consistent at the cost of the webpage sometimes
taking slightly longer to load. Note that this only works because the
footer is added to webpages before the MathJax javascript. -->
<script type="text/x-mathjax-config">
  MathJax.Hub.Config({
    "HTML-CSS": { availableFonts: ["TeX"] }
  });
</script>


</div>
</div>

</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.header').parent('thead').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open')
  });
});
</script>

<!-- code folding -->

<script>
$(document).ready(function ()  {

    // move toc-ignore selectors from section div to header
    $('div.section.toc-ignore')
        .removeClass('toc-ignore')
        .children('h1,h2,h3,h4,h5').addClass('toc-ignore');

    // establish options
    var options = {
      selectors: "h1,h2,h3",
      theme: "bootstrap3",
      context: '.toc-content',
      hashGenerator: function (text) {
        return text.replace(/[.\\/?&!#<>]/g, '').replace(/\s/g, '_').toLowerCase();
      },
      ignoreSelector: ".toc-ignore",
      scrollTo: 0
    };
    options.showAndHide = true;
    options.smoothScroll = true;

    // tocify
    var toc = $("#TOC").tocify(options).data("toc-tocify");
});
</script>

<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
