---
title: "distal_feature"
author: "Yunqi Yang"
date: "8/20/2020"
output: html_document
---

## Description: 

1. model the distal observations, tss_dist_to_snp > 100kb 

2. the feature weight, HIC and ATAC have no prediction power, predict all the observations to be negative. 

3. If adding all other functional features, including exon, intron,..., that improve the classification performance. The accuracy is still not high, $9/49\approx0.184$.

4. The important question here: 

(1). why we couldn't classify distal observations well?

(2). why intron/exon(features in closer range) better than distal features? 


## Possible next step:

1. Instead of using binary outcome y, do more exploratory analysis on PIP, to see if there's relationship between PIP and distance. 

2. Relax eQTL PIP threshold to include more into training data. To see if distal pairs increase or not. If not, this means that eQTL couldn't capture distal gene-snp association or the calculation of PIP doesn't consider distal interaction. 






```{r  }

library(ggplot2)
library(dplyr)
require(PRROC)

```


```{r  }

load('/Users/nicholeyang/Desktop/gene_level_fine_mapping/data/train_all.RData')
head(train_all_sig)

```

## process all features
```{r  }

# create unified HiC feature
hic = apply(train_all_sig[, c(11:27)], 1, sum)
train_all_sig$hic = ifelse(hic>0, 1, 0)

# ATAC data
train_all_sig$atac = ifelse(train_all_sig$correlation > 0.5, 1, 0)

# transform tss_dist_to_snp into weights 
sigma = 1e5
train_all_sig$weight = exp(-train_all_sig$tss_dist_to_snp/sigma)

```

```{r  }

dat = train_all_sig[train_all_sig$tss_dist_to_snp > 1e5, ]

sum(dat$y == 1)
sum(dat$y == 0)

```

## Create a training/test set
```{r  }

set.seed(215)

n = dim(dat)[1]
indx = sample(1:n, round(2*n/3), replace = FALSE)
train = dat[indx, ]
test = dat[-indx, ]

sum(train$y == 1)
```


## fit1: model contains weight + most of the features 
```{r  }

fit1 = glm(y ~ weight + exon + UTR5 + UTR3+ intron+ upstream, data = train, family = "binomial")
summary(fit1)

pred.probs=predict(fit1,test,type="response")
glm.pred = rep(0, length(pred.probs))


## all predicted to be negative
glm.pred[pred.probs>0.5]= 1
table(glm.pred, test$y)
```

```{r  }
# prediction accuracy
accuracy = sum(glm.pred == test$y)/length(test$y)
accuracy 
```



```{r  }

fg <- pred.probs[test$y == 1]
bg <- pred.probs[test$y== 0]

# ROC Curve    
roc <- roc.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(roc)

# PR Curve
pr <- pr.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(pr)

```



## fit2: model contains weight + exon
```{r  }

fit2 = glm(y ~ weight + exon, data = train, family = "binomial")
summary(fit2)

pred.probs=predict(fit2,test,type="response")
glm.pred = rep(0, length(pred.probs))


## all predicted to be negative
glm.pred[pred.probs>0.5]= 1
table(glm.pred, test$y)
```

```{r  }
# prediction accuracy
accuracy = sum(glm.pred == test$y)/length(test$y)
accuracy 
```

```{r  }

fg <- pred.probs[test$y == 1]
bg <- pred.probs[test$y== 0]

# ROC Curve    
roc <- roc.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(roc)

# PR Curve
pr <- pr.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(pr)

```




## fit3: model only contains weight
```{r  }

fit3 = glm(y ~ weight, data = train, family = "binomial")
summary(fit3)

pred.probs=predict(fit3,test,type="response")
glm.pred = rep(0, length(pred.probs))


## all predicted to be negative
glm.pred[pred.probs>0.5]= 1
table(glm.pred, test$y)
```

```{r  }
# prediction accuracy
accuracy = sum(glm.pred == test$y)/length(test$y)
accuracy 
```



```{r  }

fg <- pred.probs[test$y == 1]
bg <- pred.probs[test$y== 0]

# ROC Curve    
roc <- roc.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(roc)

# PR Curve
pr <- pr.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(pr)

```


## fit4: model contains weight, HiC and ATAC 
```{r  }

fit4 = glm(y ~ weight + hic + atac, data = train, family = "binomial")
summary(fit4)

pred.probs=predict(fit4,test,type="response")
glm.pred = rep(0, length(pred.probs))

## all predicted to be negative 
glm.pred[pred.probs>0.5]= 1
table(glm.pred, test$y)
```


```{r  }

# prediction accuracy
accuracy = sum(glm.pred == test$y)/length(test$y)
accuracy 


```

```{r  }

fg <- pred.probs[test$y == 1]
bg <- pred.probs[test$y== 0]

# ROC Curve    
roc <- roc.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(roc)

# PR Curve
pr <- pr.curve(scores.class0 = fg, scores.class1 = bg, curve = T)
plot(pr)

```


## More exploratory analysis:

```{r  }

## proportion of positive observations by distance
sum(dat$y == 1)/dim(dat)[1]   ## distal ones 
sum(train_all_sig$y == 1)/dim(train_all_sig)[1]

```




